v3/src/components/graph/components/scatterdots.tsx

```diff
@@ -164,6 +175,125 @@ export const ScatterDots = observer(function ScatterDots(props: PlotProps) {
+    dataTip.show(dataTipContent, event.target)
+  }, [dataTip, dataset?.name])
+
+  const connectingLinesCleanUp = useCallback(() => {
+    connectingLinesActivatedRef.current = showConnectingLines
+    // TODO: The point size needs to be made smaller when Connecting Lines are activated, and then revert to the
+    // original size when Connecting Lines are deactivated. The below makes this happen, but the rescaling should
+    // really occur in a smooth transition while the lines are fading in/out, not instantly after the lines are
+    // done fading.
+    const pointSizeMultiplier = dataDisplayModel?.pointDescription.pointSizeMultiplier
+    if (showConnectingLines && pointSizeMultiplier > .5) {
+      dataDisplayModel?.pointDescription.setPointSizeMultiplier(.5)
+    } else if (!showConnectingLines) {
+      dataDisplayModel?.pointDescription.setPointSizeMultiplier(1)
+    }
+    if (!showConnectingLines) {
+      select(connectingLinesRef.current).selectAll("path").remove()
+    }
+  }, [dataDisplayModel?.pointDescription, showConnectingLines])
+
+  const refreshConnectingLines = useCallback(() => {
+    if (!showConnectingLines && !connectingLinesActivatedRef.current) return
+    const connectingLinesArea = select(connectingLinesRef.current)
+    const curve = line().curve(curveLinear)
+    const { connectingLinesForCases } = scatterPlotFuncs(layout, dataConfiguration)
+    const connectingLines = connectingLinesForCases()
+    const parentAttr = dataset?.collections[0]?.attributes[0]
+    const parentAttrID = parentAttr?.id
+    const parentAttrName = parentAttr?.name
+    const parentAttrValues = parentAttrID && dataConfiguration?.metadata?.getCategorySet(parentAttrID)?.values
+    const cellKeys = dataConfiguration?.getAllCellKeys()
+
+    connectingLinesArea.selectAll("path").remove()
+    cellKeys?.forEach((cellKey) => {
+      // Each plot can have multiple groups of connecting lines. The number of groups is determined by the number of Y
+      // attributes or the presence of a parent attribute and the number of unique values for that attribute. If there
+      // are multiple Y attributes, the number of groups matches the number of Y attributes. Otherwise, if there's a
+      // parent attribute, the number of groups matches the number of unique values for that attribute. If there's only
+      // one Y attribute and no parent attribute, then there's a only single group. The code below builds lists of
+      // connecting lines and case IDs for each group.
+      const lineGroups: Record<string, IConnectingLineDescription[]> = {}
+      const allLineCaseIds: Record<string, string[]> = {}
+      const yAttrCount = dataConfiguration?.yAttributeIDs?.length ?? 0
+      connectingLines.forEach((lineDescription: IConnectingLineDescription) => {
+        const parentAttrValue = parentAttrID ? importValueToString(lineDescription.caseData[parentAttrID]) : undefined
```

Comment:
Do we always want string values in `caseData`? If so, you can pass `{ numeric: false }` to the call to `getCase()` in the `connectingLine()` function in `scatter-plot-utils.ts`.

Reply:
Thanks. I believe we do want string values in this case because if we use the value, it will be as a key on an object. I found that even with passing `{ numeric: false }` to `getCase()`, however, there'd still be a type error here unless I used `importValueToString` or `lineDescription.caseData[parentAttrID] as string`. Without doing that, the type of `parentAttrValue` will be `IValueType` which can be a string, number, boolean, or undefined.

Am I missing something obvious that should make that second step unnecessary? Or do you think that's potentially a problem with `getCase()`? I'm not sure. It returns a value with of type `ICase`, [the definition of which](https://github.com/concord-consortium/codap/blob/3cbd8539f1029ea79dcd075e1e681eca751a08fb/v3/src/models/data/data-set-types.ts#L12-L16) seems very reasonable, although I would expect items returned by `getCase()` to have properties whose values are either numbers or strings or maybe undefined, but not booleans.

Reply:
With the `{ numeric: false }`, _you_ know that the values are always strings, the issue is just that the compiler doesn't know it. I think I would suggest just using `String(lineDescription.caseData[parentAttrID])` as a simpler conversion than `importValueToString()`.